server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -190,6 +191,10 @@
         Setting.Property.NodeScope
     );
 
+    public static int MAX_DOC_ID_LENGTH = 512;
```

Comment:
We try hard to avoid static variables, since this hampers test-ability and leads to unstable tests. For instance in this PR, all tests that run in the same JVM after `IndexRequestTests` will run with a limit of 513.

I believe this requires doing the validation outside the request, as also explained in [#16036](https://github.com/elastic/elasticsearch/pull/16036#issuecomment-186897826).